### ISSUE TITLE
Carer connect to care team & record: Add fields to the form

### ISSUE THREAD

**Issue by MollieCourtenay** [healthlocker/healthlocker#569]
As a carer wanting to connect the to the care team of someone I care for, I am asked for me First name, Last name, and relationship to the user - so that I can be identified by the care team.

###Your details 
First name
Last name

###Person you support 
These details will be matched against their SLaM record - they must match perfectly. (REMOVE LINE BREAK)
First name
Last name
DOB
NHS number (?)
Your relationship to this person  [free text box no drop down]

Connect
 
<img width="159" alt="screen shot 2017-05-08 at 13 02 14" src="https://cloud.githubusercontent.com/assets/25012688/25803510/1962918c-33ef-11e7-93f9-1d006ee0f081.png">

#278

**Comment by tonydaly** [comment]
Hey @MollieCourtenay still confused why we need to ask the carer for their first and last name when connecting. Is there a use case where the name entered here would be different from the name they used when signing up?

**Comment by MollieCourtenay** [comment]
Hello, @tonydaly  Adding a name is optional at sign up. Someone may use a nickname. Some may not enter a name at all.

When connecting we'll ask for the actual name -  the clinician will want to know the carers actual name and relationship to the service user so they know who they are messaging.

**Comment by tonydaly** [comment]
What was the reasoning for "Name" being optional in the account?

So on this form are "First name" and "Last name" required fields? If they're entered should we override the name the user entered (or didn't enter) when they signed up? Or keep them separate? If they're not separate should we prefill these fields with the name entered in their account? 

I think it's going to be very confusing for users to come from the account page where their name is shown, to then land on another page which asks them for their name again.

**Comment by MollieCourtenay** [comment]
@tonydaly 
The ask was to simplify the sign up experience as much as possible, so we removed the need for the user to supply first and last name as not all users will use HL as a connected to SLaM user - which is the only reason we'd NEED to know their name. It was left optional as in the designs there was a personalised welcome message - (Welcome 'name') which wasn't built.

To connect we need to match the health record.
So if we ask them to provide a first and last name in sign up - and for example "Matt"  "Smith" is entered. Then they try to connect, but on the record its "Matthew" "Smith", it wouldn't match.
 
I guess we could remove any ask for a name in sign up - and only ask for first name and last name if the user wants to connect. this may be a better way round. What do you think?

Yes we should override with the details used to connect to SLaM over what was entered at sign up - if we stick with it as it is..

**Comment by tonydaly** [comment]
Confused about one thing:

> To connect we need to match the health record.

But in our previous conversations you mentioned:

> Carers may or may not be in ePJS but HL is not interested in that.

So if there's no carer record in ePJS, what are we checking against? We currently don't model any carer information in ePJS.

---

I feel that "Name" is always going to be an essential piece of information (especially if we want to start communicating with users), so I'd definitely keep it. 

Overriding the name works fine, but could we standardise on the fields we ask for? On this page it's "First name" + "Last name", but on signup/account it's just one "Name" field (the single, existing field will be much easier).

**Comment by MollieCourtenay** [comment]
@tonydaly    Yeah, you're right carers names we don't check - but carers and SUs use the same sign up form. so we need it to work for both user types.

I don't think users will always put in the correct full name on sign up. and so we will need to ask again when they connect...

We could ask for first and last name in sign up, then pre-populate those fields in the connect form, giving users the option to correct what they had previously entered.

Single field is easier for who? users or you developing it?
Are you able to check the SUs name against ePJS DB with one name field? in ePJS it is First and Last, that's why it has been designed like that.

**Comment by katbow** [comment]
I'm going to be splitting this into a few technical issues, but here's my overall plan for this:

* Add first & last name to HL db
* Add first & last name fields to carer connecting form
* Update this in HL when a carer connects
* Update from ePJS when SLaM SU connects & display in slam_connected account profile
* Change sign up form to accept first & last name
* Update name anywhere it's displayed/used to first name + last name
* Update seeds? May only need to update the known users for slam connecting/slam connecting
* Delete name field from db (question about what to do with the data which I'll put into more detail in another issue).

**Comment by katbow** [comment]
@MollieCourtenay can this be the issue for name instead since that's what the conversation focussed on so far? Might need to do some jumping back & forth otherwise.

**Comment by MollieCourtenay** [comment]
@katbpw thought this was what was agreed Tuesday. To edit to use this for relationship and create one issue for everywhere where the name fields need to be first and last. 

But whatever makes it easier for you to work on. Go for it. :)

**Comment by MollieCourtenay** [comment]
@katbow

**Comment by iteles** [comment]
Deprioritised by Nicola Byrne in favour of https://github.com/healthlocker/healthlocker/issues/650.
